plugins/trace_api_plugin/store_provider.cpp

```diff
@@ -33,6 +34,7 @@ namespace eosio::trace_api {
       _slice_directory.find_or_create_index_slice(slice_number, true, index);
       auto le = metadata_log_entry { lib_entry_v0 { .lib = lib }};
       append_store(le, index);
+      _slice_directory.cleanup_old_slices(lib);
```

Comment:
I think this is fine for now however, we should add a backlog task to process this on a low priority timer instead of in the critical path of the writing process.  I don't think it will do work often enough to warrant a change prior to the initial release.

Reply:
I had thought about that, but figured that this is only on lib write and since I cache the last deleted it is only on one lib write per stride, all other ones will just do a simple logic check.